src/v/resource_mgmt/storage.cc

```diff
@@ -280,11 +280,13 @@ eviction_policy::collect_reclaimable_offsets() {
     co_await ss::max_concurrent_for_each(
       partitions.begin(), partitions.end(), 20, [&res, cfg](const auto& p) {
           auto log = p->log();
+          auto size = log->size_bytes();
```

Comment:
I don't quite recall the nuances here: would something like `offsets.size` be worth considering? I could be misremembering but if it represents a collectible size estimate, it seems like an intuitive choice (also wondering if it's possible for there to be large partitions with not much reclaimable space... maybe configured with larger segments?)

Reply:
>I don't quite recall the nuances here: would something like offsets.size be worth considering?

The idea here would be to report size not as the size of the partition but the size of the reclaimable space. This would capture the case where uploads weren't functioning for a partition and the max collectible offset wasn't progressing. Not sure how much difference this would make in practice but it does kind of feel like it is more closely related to the goal. Gotta think about it